examples/graph-analytics-serverless-spark.ipynb

```diff
@@ -0,0 +1,360 @@
+    "Working with a remote Spark cluster will work similarly."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "from pyspark.sql import SparkSession\n",
+    "\n",
+    "os.environ[\"JAVA_HOME\"] = \"/home/max/.sdkman/candidates/java/current\"\n",
+    "\n",
+    "spark = SparkSession.builder.master(\"local[4]\").appName(\"GraphAnalytics\").getOrCreate()\n",
+    "\n",
+    "# Enable Arrow-based columnar data transfers\n",
+    "spark.conf.set(\"spark.sql.execution.arrow.pyspark.enabled\", \"true\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Aura API credentials\n",
+    "\n",
+    "The entry point for managing GDS Sessions is the `GdsSessions` object, which requires creating [Aura API credentials](https://neo4j.com/docs/aura/api/authentication)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "from graphdatascience.session import AuraAPICredentials, GdsSessions\n",
+    "\n",
+    "# you can also use AuraAPICredentials.from_env() to load credentials from environment variables\n",
+    "api_credentials = AuraAPICredentials(\n",
+    "    client_id=os.environ[\"CLIENT_ID\"],\n",
+    "    client_secret=os.environ[\"CLIENT_SECRET\"],\n",
+    "    # If your account is a member of several project, you must also specify the project ID to use\n",
+    "    project_id=os.environ.get(\"PROJECT_ID\", None),\n",
+    ")\n",
+    "\n",
+    "sessions = GdsSessions(api_credentials=api_credentials)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Creating a new session\n",
+    "\n",
+    "A new session is created by calling `sessions.get_or_create()` with the following parameters:\n",
+    "\n",
+    "* A session name, which lets you reconnect to an existing session by calling `get_or_create` again.\n",
+    "* The session memory. \n",
+    "* The cloud location.\n",
+    "* A time-to-live (TTL), which ensures that the session is automatically deleted after being unused for the set time, to avoid incurring costs.\n",
+    "\n",
+    "See the API reference [documentation](https://neo4j.com/docs/graph-data-science-client/current/api/sessions/gds_sessions/#graphdatascience.session.gds_sessions.GdsSessions.get_or_create) or the manual for more details on the parameters."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from datetime import timedelta\n",
+    "\n",
+    "from graphdatascience.session import CloudLocation, SessionMemory\n",
+    "\n",
+    "# Create a GDS session!\n",
+    "gds = sessions.get_or_create(\n",
+    "    # we give it a representative name\n",
+    "    session_name=\"bike_trips\",\n",
```

Comment:
i read about bike trips just today in our logs. nice to learn its your workload 👀